### PR TITLE
[mergeTreeTemporalReduction] add module

### PR DESCRIPTION
**Pull request by MatPont**
This PR adds the temporal reduction module for merge trees.

It allows to subsampling a sequence of merge trees by greedily remove trees in the original sequence that can be accurately reconstructed with the geodesic computation.

Best.

### CODE REVIEW

File: core/base/mergeTreeTemporalReductionDecoding/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ttk_add_base_library(mergeTreeTemporalReductionDecoding
+  SOURCES
+    MergeTreeTemporalReductionDecoding.cpp
+  HEADERS
+    MergeTreeTemporalReductionDecoding.h
+  DEPENDS
+    triangulation
+    ftmTree
```
Comment (julien-tierny) [inline core/base/mergeTreeTemporalReductionDecoding/CMakeLists.txt]: do you need this dep for the `ftmTreeUtils`? (traversal and such)

File: core/base/mergeTreeTemporalReductionEncoding/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+ttk_add_base_library(mergeTreeTemporalReductionEncoding
+  SOURCES
+    MergeTreeTemporalReductionEncoding.cpp
+  HEADERS
+    MergeTreeTemporalReductionEncoding.h
+  DEPENDS
+    triangulation
+    ftmTree
```
Comment (julien-tierny) [inline core/base/mergeTreeTemporalReductionEncoding/CMakeLists.txt]: same remark (cf decoder)

File: core/vtk/ttkMergeTreeTemporalReductionDecoding/ttk.module [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  ttkMergeTreeTemporalReductionDecoding
+SOURCES
+  ttkMergeTreeTemporalReductionDecoding.cpp
+HEADERS
+  ttkMergeTreeTemporalReductionDecoding.h
+DEPENDS
+  mergeTreeTemporalReductionDecoding
+  ttkAlgorithm
+  ttkPlanarGraphLayout
+  ftmTree
```
Comment (julien-tierny) [inline core/vtk/ttkMergeTreeTemporalReductionDecoding/ttk.module]: same remark as above.

File: core/vtk/ttkMergeTreeTemporalReductionEncoding/ttk.module [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  ttkMergeTreeTemporalReductionEncoding.cpp
+HEADERS
+  ttkMergeTreeTemporalReductionEncoding.h
+  ttkMergeTreeTemporalReductionUtils.h
+DEPENDS
+  mergeTreeTemporalReductionEncoding
+  ttkAlgorithm
+  ttkFTMTree
+  ttkPlanarGraphLayout
+  dimensionReduction
```
Comment (julien-tierny) [inline core/vtk/ttkMergeTreeTemporalReductionEncoding/ttk.module]: I'm curious about that dep...
do you use it?

File: paraview/xmls/MergeTreeTemporalReductionDecoding.xml [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+            <!-- OUTPUT PARAMETER WIDGETS -->
+                <OutputPort name="Reconstructed Sequence" index="0" id="port0" />
+                <OutputPort name="Matching" index="1" id="port1" />
+
+      <!-- DEBUG -->
+      ${DEBUG_WIDGETS}
+
+      <!-- MENU CATEGORY -->
+      <Hints>
+        <ShowInMenu category="TTK - Ensemble Scalar Data" />
```
Comment (julien-tierny) [inline paraview/xmls/MergeTreeTemporalReductionDecoding.xml]: TTK - Time-varying Scalar Data

File: paraview/xmls/MergeTreeTemporalReductionEncoding.xml [hunk 6]
```diff
@@ -0,0 +1,460 @@
...
+            <!-- OUTPUT PARAMETER WIDGETS -->
+                <OutputPort name="Key Frames" index="0" id="port0" />
+                <OutputPort name="Metadata" index="1" id="port1" />
+
+      <!-- DEBUG -->
+      ${DEBUG_WIDGETS}
+
+      <!-- MENU CATEGORY -->
+      <Hints>
+        <ShowInMenu category="TTK - Ensemble Scalar Data" />
```
Comment (julien-tierny) [inline paraview/xmls/MergeTreeTemporalReductionEncoding.xml]: "TTK - Time-varying Scalar Data"

**Review by julien-tierny — CHANGES_REQUESTED** [review]
a couple of edit suggestions and a few questions/remarks.

File: core/vtk/ttkMergeTreeTemporalReductionEncoding/ttk.module (hunk 4 above)
Comment (julien-tierny) [reply core/vtk/ttkMergeTreeTemporalReductionEncoding/ttk.module]: (to remove)

File: core/base/mergeTreeTemporalReductionDecoding/CMakeLists.txt (hunk 1 above)
Comment (MatPont) [reply core/base/mergeTreeTemporalReductionDecoding/CMakeLists.txt]: Removed in the next commit, thanks!

File: core/base/mergeTreeTemporalReductionEncoding/CMakeLists.txt (hunk 2 above)
Comment (MatPont) [reply core/base/mergeTreeTemporalReductionEncoding/CMakeLists.txt]: Removed in the next commit, thanks!

File: core/vtk/ttkMergeTreeTemporalReductionDecoding/ttk.module (hunk 3 above)
Comment (MatPont) [reply core/vtk/ttkMergeTreeTemporalReductionDecoding/ttk.module]: Removed in the next commit, thanks!

File: core/vtk/ttkMergeTreeTemporalReductionEncoding/ttk.module (hunk 4 above)
Comment (MatPont) [reply core/vtk/ttkMergeTreeTemporalReductionEncoding/ttk.module]: Removed in the next commit, thanks!

File: paraview/xmls/MergeTreeTemporalReductionDecoding.xml (hunk 5 above)
Comment (MatPont) [reply paraview/xmls/MergeTreeTemporalReductionDecoding.xml]: Done in the next commit, thanks!

File: paraview/xmls/MergeTreeTemporalReductionEncoding.xml (hunk 6 above)
Comment (MatPont) [reply paraview/xmls/MergeTreeTemporalReductionEncoding.xml]: Done in the next commit, thanks!

**Review by julien-tierny — COMMENTED** [review]
Looks good, thanks!

**Review by julien-tierny — APPROVED** [review]

**Pull request merged**